PrivateTalkApp/Home/ViewModel/CalendarViewModel.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  CalendarViewModel.swift
+//  PrivateTalkApp
+//
+//  Created by 都甲裕希 on 2024/08/22.
+//
+
+import Foundation
+
+// MARK: - Calendar ViewModel
+@MainActor
+final class CalendarViewModel: ObservableObject {
+    
+    private struct Constants {
+        static let FULL_DATE_FORMAT = "yyyy-MM-dd HH:mm:ss"
+    }
+    
+    // カレンダーのModel
+    @Published var calendarModel: CalendarModel?
+    // WorlTimeAPIの世界時刻情報を取得するために使用するService
+    private let worldTimeService = WorldTimeService()
+    
+    /// 年月文字列をセット
+    /// - parameter date: セットしたいDate
+    func setDisplayDate(date: Date?) {
+        Task { [weak self] in
+            guard let self = self else {
+                return
+            }
+            self.calendarModel = CalendarModel(date: date)
+        }
+    }
+    
+    /// 今日ボタンを押下された際の処理
+    func tapTodayButton() {
+        Task {
+            do {
+                // 現在の時刻をUTC文字列で取得
+                let response = try await self.worldTimeService.fetchWorldTimeApi()
+                // UTC文字列をUTCのDateに変換
+                let currentUtcDate = DateUtilities.convertStringToUtcDate(dateString: response.datetime,
+                                                                          format: Constants.FULL_DATE_FORMAT)
+                // UTCのDateからローカルタイムゾーンの年月文字列をセットする
+                self.setDisplayDate(date: currentUtcDate)
```

Comment:
viewModel全体が@MainActorだからこれでいけるはず
```suggestion
                self.calendarModel = CalendarModel(date: date)
```

Reply:
この実装の意図としては、メンテナンス性的に複数箇所でcalendarModelに代入するのが嫌で、setするメソッドを作成してそれを呼ぶようにしました。

Reply:
なるほど。
そしたら、ここのTaskだけはずしな。いらないはず
https://github.com/tokouyuuki/PrivateTalkApp/blob/c98509f73c5bc367999301e9efc541cd365c8de0/PrivateTalkApp/Home/ViewModel/CalendarViewModel.swift#L26-L31

Reply:
ここUI更新中に、次の更新がかかっちゃってエラー起こるのよね
`Publishing changes from within view updates is not allowed, this will cause undefined behavior.`
Dispatchque.main.asyncとかの方が良い？

Reply:
まじか、じゃあこのままでいいと思います